source/engine/physics2/jolt/JobSystem.cpp

```diff
@@ -0,0 +1,90 @@
+#include "JobSystem.h"
+
+#include "ncengine/task/AsyncDispatcher.h"
+#include "ncutility/NcError.h"
+
+#include "Jolt/Jolt.h"
+#include "Jolt/Core/FixedSizeFreeList.h"
+#include "Jolt/Core/JobSystemWithBarrier.h"
+#include "Jolt/Physics/PhysicsSettings.h"
+
+namespace
+{
+// JobSystem implementation that routes jobs to Taskflow
+class TaskRouter : public JPH::JobSystemWithBarrier
+{
+    public:
+        TaskRouter(const nc::task::AsyncDispatcher& dispatcher)
+            : JPH::JobSystemWithBarrier{JPH::cMaxPhysicsBarriers},
```

Comment:
In the sample implementation Jolt provides for their `JobSystemWithBarrier`, they initialize their job system with a number of threads for their thread pool. Don't we want to do the same and keep ours in sync with what the user specifies in EngineSettings?

https://github.com/jrouwe/JoltPhysics/blob/2292c415a9a94cd7b4706de6ac8f24c9b5af74a5/Jolt/Core/JobSystemThreadPool.h#L31